File: Rust/day-01/src/bin/part2.rs
fn main() {
    let input = include_str!("input2.txt");
    let output = part2(input);
    dbg!(output);
}

fn part2(input: &str) -> u32 {
    input
        .split_terminator('\n')
        .map(|entry| {
            entry
                .replace("one", "o1e")
                .replace("two", "t2o")
                .replace("three", "t3e")
                .replace("four", "f4r")
                .replace("five", "f5e")
                .replace("six", "s6x")
                .replace("seven", "s7n")
                .replace("eight", "e8t")
                .replace("nine", "n9e")
        })
        .filter_map(|entry| {
            let mut start = 0;
            let mut end = 0;
            for c in entry.chars() {
                if c.is_numeric() {
                    if start == 0 {
                        start = c.to_digit(10).unwrap() * 10;
                        end = c.to_digit(10).unwrap();
                    } else {
                        end = c.to_digit(10).unwrap();
                    }
                }
            }
            Some(start + end)
        })
        .sum()
}

#[cfg(test)]
mod test {
    use super::*;
    #[test]
    fn example_test() {
        let input = "two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen
";
        assert_eq!(part2(input), 281);
    }
}

File: Python/Day-01/part2.py
import re
def main():
    with open("input.txt", "r") as file:
        string_numbers = {
            "one": 1, 
            "two": 2,
            "three": 3, 
            "four": 4, 
            "five": 5, 
            "six": 6, 
            "seven": 7, 
            "eight": 8, 
            "nine": 9
            }
        total_calibarations = 0
        content = file.read()
        calibarations = content.split('\n')
    
        for cal in calibarations:
            first_number = -1
            last_number = -1 
            while (len(cal)):
                for num in string_numbers.keys():
                    if re.search(f"^{num}", cal):
                        if first_number == -1:
                            first_number = string_numbers[num] * 10
                            last_number = string_numbers[num]
                        last_number = string_numbers[num]
                if cal[0].isnumeric():
                    if first_number == -1:
                            first_number = int(cal[0]) * 10
                            last_number = int(cal[0])
                    last_number = int(cal[0])
                cal = cal[1:]

            total_calibarations += first_number + last_number

        print(total_calibarations)







if __name__ == "__main__":
    main()